### PR TITLE
feature: add metals.copy-worksheet-output

### PR DESCRIPTION
**Pull request by JakDar**
Added  metals.copy-worksheet-output introduced (I think) in [this metals PR](https://github.com/scalameta/metals/pull/2290).
Implementation is based on [PR from coc-emacs](https://github.com/scalameta/coc-metals/pull/360).

Happy to fix code if there are any issues 😄

### CODE REVIEW

File: lsp-metals.el [hunk 1]
```diff
@@ -231,6 +231,15 @@ change it again."
   (interactive)
   (lsp-send-execute-command "reset-choice" ()))
 
+(defun lsp-metals-copy-worksheet-output ()
+  "Copy worksheet with evaluated results as comments."
+  (interactive)
+  (let ((command-result (lsp-send-execute-command "metals.copy-worksheet-output" (lsp--buffer-uri))))
+    (if (ht-get command-result "value")
```
Comment (yyoncho) [inline lsp-metals.el:238]: we currently support running with both plists/hts as a json deserialization target. This means that you should either use lsp-get or use lsp-interface like this: https://github.com/emacs-lsp/lsp-metals/blob/3dd96c1a72012735ac2874b89461f7d65b58fa8a/lsp-metals-protocol.el

Comment (JakDar) [reply lsp-metals.el]: fixed ;)

File: lsp-metals.el [hunk 2]
```diff
@@ -231,6 +231,15 @@ change it again."
   (interactive)
   (lsp-send-execute-command "reset-choice" ()))
 
+(defun lsp-metals-copy-worksheet-output ()
+  "Copy worksheet with evaluated results as comments."
+  (interactive)
+  (let ((command-result (lsp-send-execute-command "metals.copy-worksheet-output" (lsp--buffer-uri))))
+    (if (lsp-get command-result :value)
```
Comment (yyoncho) [inline lsp-metals.el:238]: nit: if -> when and remove and you can remove the progn. After this it looks good to me from elisp perspective, and I will let @kurnevsky to test the metals part and merge it.

Comment (JakDar) [reply lsp-metals.el]: yup, that looks cleaner.

**Comment by kurnevsky** [comment]
Thanks!

**Pull request merged**